frontend/src/preferences/preferenceApi.ts

```diff
@@ -46,4 +46,20 @@ export function setLocalApiKey(key: string): void {
 
 export function setLocalRecentDocs(doc: Doc): void {
   localStorage.setItem('DocsGPTRecentDocs', JSON.stringify(doc));
+  let namePath = doc.name;
+  if (doc.language === namePath) {
+    namePath = '.project';
+  }
+
+  const docPath =
+    doc.language + '/' + namePath + '/' + doc.version + '/' + doc.model;
+  fetch('https://docsgpt.arc53.com/api/docs_check', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      docs: docPath,
+    }),
+  }).then((response) => response.json());
```

Comment:
i feel like the failed HTTP request arent raising rejected promise properly. i checked with random api key on my local.

Reply:
```
let namePath = selectedDocs.name;
  if (selectedDocs.language === namePath) {
    namePath = '.project';
  }

  const docPath =
    selectedDocs.language +
    '/' +
    namePath +
    '/' +
    selectedDocs.version +
    '/' +
    selectedDocs.model;

  return fetch('https://docsgpt.arc53.com/api/answer', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      question: question,
      api_key: apiKey,
      embeddings_key: apiKey,
      history: localStorage.getItem('chatHistory'),
      active_docs: docPath,
    }),
  })
    .then((response) => {
      if (response.ok) {
        return response.json();
      } else {
        Promise.reject(response);
      }
    })
    .then((data) => {
      const result = data.answer;
      return { answer: result, query: question, result };
    });
```

Alternate way which rejects the promise when HTTP req doesnt succeed.
fetch promises are successful even http request fails, so we gotta manually check `response.ok`

Reply:
BTW i am not confident if this covers all conditions. Please test and make necessary modifications.